tests/test_live.py

```diff
@@ -0,0 +1,1428 @@
+
+    proc = subprocess.Popen(
+        cmd,
+        universal_newlines=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE)
+    stdout, stderr = proc.communicate()
+
+    if proc.returncode != 0:
+        print(stderr)
+
+    lines = []  # type: List[str]
+    for line in stdout.split('\n'):
+        line = line.strip()
+        # empty line
+        if line == '':
+            continue
+        # subdirectory matching resolved wildcard *
+        if line.endswith('/:'):
+            continue
+        lines.append(line)
+
+    return lines
+
+
+def call_gcs_client_ls(client: gswrap.Client,
+                       path: str,
+                       recursive: bool = False) -> List[str]:
+
+    try:
+        list_ls = client.ls(gcs_url=path, recursive=recursive)
+    except google.api_core.exceptions.GoogleAPIError as err:
+        print(err)
+        return []
+
+    return list_ls
+
+
+class TestLS(unittest.TestCase):
+    def setUp(self):
+        # pylint: disable=fixme
+        # TODO(snaji): remove warning filters
+        if NO_WARNINGS:
+            warnings.filterwarnings(
+                "ignore",
+                category=UserWarning,
+                message=
+                "Your application has authenticated using end user credentials "
+                "from Google Cloud SDK.*")
+            warnings.filterwarnings(
+                "ignore",
+                category=ResourceWarning,
+                message="unclosed.*<ssl.SSLSocket.*>")
+
+        self.client = gswrap.Client(bucket_name=TEST_GCS_BUCKET)
+        self.bucket_prefix = str(uuid.uuid4())
+        gcs_test_setup(prefix=self.bucket_prefix)
+
+    def tearDown(self):
+        gcs_test_teardown(prefix=self.bucket_prefix)
+
+    def test_gsutil_ls_non_recursive(self):
+        test_cases = [
+            # no wildcards
+            "gs://bucket-inexistent",
+            "gs://{}".format(TEST_GCS_BUCKET_NO_ACCESS),
+            "gs://{}".format(TEST_GCS_BUCKET[:-1]),
+            "gs://{}".format(TEST_GCS_BUCKET),
+            "gs://{}/".format(TEST_GCS_BUCKET),
+            "gs://{}/{}".format(TEST_GCS_BUCKET, self.bucket_prefix[:-1]),
+            "gs://{}/{}".format(TEST_GCS_BUCKET, self.bucket_prefix),
+            "gs://{}/{}/".format(TEST_GCS_BUCKET, self.bucket_prefix),
+            "gs://{}/{}/d".format(TEST_GCS_BUCKET, self.bucket_prefix),
+            "gs://{}/{}/d1".format(TEST_GCS_BUCKET, self.bucket_prefix),
+            "gs://{}/{}/d1/".format(TEST_GCS_BUCKET, self.bucket_prefix)
+        ]
+
+        for i, test_case in enumerate(test_cases):
+            print("{}/{}".format(i + 1, len(test_cases)))
+            start_gsutil = time.time()
+            list_gsutil = call_gsutil_ls(path=test_case)
+            end_gsutil = time.time()
+            start_gcs = time.time()
+            list_gcs = call_gcs_client_ls(client=self.client, path=test_case)
+            end_gcs = time.time()
+            if VERBOSE:
```

Comment:
I'm not sure if I like this VERBOSE part. Let's leave it as is and discuss it with Marko when everything else is ready